src/python/WMCore/Credential/Proxy.py

```diff
@@ -159,6 +159,8 @@ def setEnv(self, cmd):
         if self.uisource is not None and len(self.uisource) > 0:
             ret = 'source ' + self.uisource + ' && ' + ret
         if self.cleanEnvironment:
+            # Need to escape ' in cmd
+            ret = ret.replace("'", "'\\''")
```

Comment:
This is correct, see : http://wiki.bash-hackers.org/syntax/quoting
`r''` will only make one backslash less
```
In practise that means, to produce a text like Here's my test… as a single-quoted string, you have to leave and re-enter the single quoting to get the character "'" as literal text:

# WRONG
echo 'Here's my test...'

# RIGHT
echo 'Here'\''s my test...'

# ALTERNATIVE: It's also possible to mix-and-match quotes for readability:
echo "Here's my test"
```